### DISCUSSION TITLE
Snapshot and migrations

### DISCUSSION THREAD

**Discussion by keesvanbemmel** [directus/directus#8968] [Archive]
Hi all!

Just wondering: what should I run first? Migrations or applying a schema snapshot?

I thought I'd always run migrations first and after that I'll run the schema apply. However, what if I have a custom migration that adds an index on a column from a collection/table added by schema apply?

:) Chicken egg problem? Or is there a simple answer?

Cheers!

**Comment by joselcvarela** [comment]
Hello @keesvanbemmel 
In my opinion, you should run first the schema apply, so directus migrations could be applied on top of  your current schema.
But, I think you need to prepare your migrations in case it has already the modifications. For example, `createTableIfNotExists` instead of `createTable`.

**Reply by keesvanbemmel** [reply]
And just to be sure: the schema snapshot doesn't contain database indexes right?

**Reply by rijkvanzanten** [reply]
That's correct 👍🏻 Only foreign key constraints and unique indexes are tracked

**Comment by keesvanbemmel** [comment]
Cool! Just added a custom migration to add the compound unique constraint
and index. Works like a charm.
